src/Catalog/Monitoring/ValidationTestCollector.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Result;
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test;
+using NuGet.Services.Metadata.Catalog.Persistence;
+using Microsoft.Extensions.Logging;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    /// <summary>
+    /// Runs a <see cref="PackageValidationTestRunner"/> against collected packages.
+    /// </summary>
+    public class ValidationTestCollector : SortingIdVersionCollector
+    {
+        public ValidationTestCollector(Storage auditingStorage, PackageValidationTestRunner packageValidationTestRunner, Uri index, Func<HttpMessageHandler> handlerFunc = null, ILogger logger = null) : base(index, handlerFunc)
+        {
+            _auditingStorage = auditingStorage;
+            _packageValidationTestRunner = packageValidationTestRunner;
+            _logger = logger;
+        }
+        
+        public IEnumerable<PackageValidationTestResult> PackageValidationResults => new List<PackageValidationTestResult>();
+
+        protected override async Task ProcessSortedBatch(CollectorHttpClient client, KeyValuePair<Tuple<string, string>, IList<JObject>> sortedBatch, JToken context, CancellationToken cancellationToken)
```

Comment:
I didn't want to use `PackageIdentity` because it parses the version into a `NuGetVersion`, which could throw unexpected exceptions. I would prefer to keep it as a tuple of strings for this reason.